python/pyarrow/tests/test_dataset.py

```diff
@@ -351,3 +353,62 @@ def test_file_system_discovery(mockfs, paths_or_selector):
     assert isinstance(table, pa.Table)
     assert len(table) == 10
     assert table.num_columns == 4
+
+
+@pytest.mark.parquet
+def test_open_dataset_single_file(tempdir):
```

Comment:
@kszucs in the other dataset tests you have been using a MockFilesystem, while I used here pytest tempdir fixture, as we do in other tests. Was there a specific reason to use the mock filesystem (or that I should also use it here)?

Reply:
Less IO, but it's fine to use the local filesystem.